byterun/extern.c

```diff
@@ -383,6 +383,8 @@ static void writecode64(int code, intnat val)
 
 /* Marshal the given value in the output buffer */
 
+int caml_extern_allow_out_of_heap = 0;
```

Comment:
Would it make sense to make this a new flag affecting marshalling https://github.com/mshinwell/ocaml/blob/11aa1912798ce4b455068205d587a10056bfd4bb/byterun/extern.c#L40-L45

Alternatively, would it make sense to expose ways to modify this in the OCaml API of `Marhsal`?